ce/ce/util/curly-replacements.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export function replaceCurlyBraces(subject: string, properties: Map<string, string>) {
+  // One of these tokens:
+  // {{
+  // }}
+  // {variable}
+  // {
+  // }
+  // (anything that has no {}s)
+  const tokenRegex = /{{|}}|{([^}]+)}|{|}|[^{}]+/y;
+  const resultElements : Array<string> = [];
+  for (;;) {
+    const thisMatch = tokenRegex.exec(subject);
+    if (thisMatch === null) {
+      return resultElements.join('');
+    }
+
+    const wholeMatch = thisMatch[0];
+    if (wholeMatch === '{{' || wholeMatch === '{') {
```

Comment:
In `api_stable_format()`, unmatched `{` or `}` are illegal. I think this is minor enough to skip for now, but we should consider it as a bug for the future.

Reply:
Fixed in https://github.com/microsoft/vcpkg-tool/pull/672/commits/b3607be74b142a27a1605e6d55551cb328e7636d